server/controllers/finance/reports/shared.js

```diff
@@ -0,0 +1,85 @@
+  { field : 'invoice_uuid', displayName : 'FORM.LABELS.INVOICE' },
+  { field : 'service_id', displayName : 'FORM.LABELS.SERVICE' },
+  { field : 'inventory_uuid', displayName : 'FORM.LABELS.INVENTORY' },
+
+  { field : 'flux_id', displayName: 'STOCK.FLUX' },
+  { field : 'status', displayName: 'STOCK.STATUS.LABEL' },
+  { field : 'hospital_no', displayName : 'FORM.LABELS.HOSPITAL_NO' },
+  { field : 'patient_group_uuid', displayName : 'PATIENT_GROUP.PATIENT_GROUP' },
+
+  { field : 'entry_date_from', displayName: 'STOCK.ENTRY_DATE', comparitor: '>', isDate : true },
+  { field : 'entry_date_to', displayName: 'STOCK.ENTRY_DATE', comparitor: '<', isDate : true },
+  { field : 'expiration_date_from', displayName: 'STOCK.EXPIRATION_DATE', comparitor: '>', isDate : true },
+  { field : 'expiration_date_to', displayName: 'STOCK.EXPIRATION_DATE', comparitor: '<', isDate : true },
+  { field : 'is_exit', displayName : 'STOCK.OUTPUT' },
+  { field : 'label', displayName : 'STOCK.LOT' },
+  { field : 'depot_uuid', displayName : 'STOCK.DEPOT' },
+
+  { field : 'description', displayName : 'FORM.LABELS.DESCRIPTION' },
+  { field : 'entity_uuid', displayName : 'FORM.LABELS.ENTITY' },
+
+  { field : 'display_name', displayName : 'FORM.LABELS.NAME' },
+  { field : 'sex', displayName : 'FORM.LABELS.GENDER' },
+  { field : 'code', displayName : 'FORM.LABELS.CODE' },
+  { field : 'dateBirthFrom', displayName : 'FORM.LABELS.DOB', comparitor : '>', isDate : true },
+  { field : 'dateBirthTo', displayName : 'FORM.LABELS.DOB', comparitor : '<', isDate : true },
+  { field : 'dateEmbaucheFrom', displayName : 'FORM.LABELS.DATE_EMBAUCHE', comparitor : '>', isDate : true },
+  { field : 'dateEmbaucheTo', displayName : 'FORM.LABELS.DATE_EMBAUCHE', comparitor : '<', isDate : true },
+  { field : 'grade_id', displayName : 'FORM.LABELS.GRADE' },
+  { field : 'fonction_id', displayName : 'FORM.LABELS.FUNCTION' },
+  { field : 'service_id', displayName : 'FORM.LABELS.SERVICE' },
+  { field : 'type_ids', displayName : 'FORM.LABELS.TRANSACTION_TYPE' },
+
+  { field : 'reference', displayName : 'FORM.LABELS.REFERENCE' },
+  { field : 'reversed', displayName : 'CASH.REGISTRY.REVERSED_RECORDS' },
+  { field : 'limit', displayName : 'FORM.LABELS.LIMIT' },
+  { field : 'period', displayName : 'TABLE.COLUMNS.PERIOD', isPeriod : true },
+  { field : 'custom_period_start', displayName : 'PERIODS.START', isDate : true, comparitor : '>' },
+  { field : 'custom_period_end', displayName : 'PERIODS.END', isDate : true, comparitor : '<' },
+];
+
+_.sortBy(columns, c => c.field);
+
+// translation key mappings for dynamic filters
+// Basically, to show a pretty filter bar, this will translate URL query string
+// into human-readable text to be placed in the report, showing the properties
+// filtered on.
+function formatFilters(qs) {
+  return columns.filter(column => {
+    const value = qs[column.field];
+
+    if (!_.isUndefined(value)) {
```

Comment:
Hmmm ... the flexibility of this logic is problematic. It seems like no error will be thrown if the client send an unknown filter key. silently ignoring exception is so ... unsafe don't you think?

Reply:
That is actually how filter parser works.  For example "renderer" is not a filter, but still exists on the query string.  So, will ignore the non-filter keys.  It's like setting `autoParseStatements` to false for the renderer.